### ISSUE TITLE
Arrow keys stop working after running dotnet, but can be fixed by running another command

### ISSUE THREAD

**Issue by SamWindell** [nushell/nushell#6145]
### Describe the bug

If I run dotnet, the arrow keys no longer work. However, they will start working again if another command run, or enter is pressed.

This is presumably related to the issue #5220.

This is happening on my Linux machine and my Mac machine. I'm running the latest version of nu from main. It's also happening on all of the terminal applications that I have.

My config.nu is directly copied from the repos default_config.nu.

### How to reproduce

1. Check the arrow keys are working by cycling through the history
2. Run `dotnet --version`
3. Try cycling through history again with the arrow keys - not working
4. Run another command, e.g. `ls`
5. Notice the arrow keys work again

### Expected behavior

I would expect the arrow keys to always work

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.66.1                                          |
| branch             | master                                          |
| commit_hash        |                                                 |
| build_os           | linux-x86_64                                    |
| build_target       | x86_64-unknown-linux-gnu                        |
| rust_version       | rustc 1.62.0 (a8314ef7d 2022-06-27)             |
| rust_channel       | stable-x86_64-unknown-linux-gnu                 |
| cargo_version      | cargo 1.62.0 (a748cf5a3 2022-06-08)             |
| pkg_version        | 0.66.1                                          |
| build_time         | 2022-07-27 10:11:49 +01:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

**Comment by fdncred** [comment]
do you have `shell_integration: true` set in your config.nu?

**Comment by SamWindell** [comment]
> do you have `shell_integration: true` set in your config.nu?

Yes, my config file is exactly the default_config.nu and it does indeed contain that line.

**Comment by fdncred** [comment]
Hmm, if you do this, once it's messed up, does it fix it?
```
> $"(ansi csi)?1l"
```

**Comment by SamWindell** [comment]
That seems to work, but I've just discovered that pressing enter with or without any text in the prompt fixes the issue. 

Also just added some clarification to my original post: I have been getting this in multiple terminal applications.

**Comment by fdncred** [comment]
We're passing that ansi sequence before we execute the command line. Maybe we should execute it after we execute the command line? or before and after, who knows? LOL

**Comment by fdncred** [comment]
If you can, maybe you can see if PR #6153 fixes the issue for you?

**Comment by SamWindell** [comment]
Thanks, I'll try out the PR tomorrow!

**Comment by SamWindell** [comment]
No luck I'm afraid, nothing changed. I appreciate your fast response to this though.

I'm not used to rust's build system, so just to clarify: I cloned your fork, checked out out the move_application_reset_post_execute branch, ran `cargo build --release --workspace --features=extra && cargo run --release --features=extra` and then checked that the only nu binary on my system (and the one that's running) was the one just created.

**Comment by fdncred** [comment]
Here's my general process.
1. clone the repo if you don't have it already.
2. Since I already have the repo I make sure I'm on the latest main with `git checkout main`, `git fetch upstream`, `git merge upstream/main` 
3. For tests like this, I just Github's command line app and do `gh pr checkout 6153`
4. Then `cargo run --features=extra`, this is in debug but works fine and builds quicker

Your steps probably work fine though.

regarding the issue, if #6153 does not fix the issue then there's some other problem that someone will have to investigate and identify.

**Comment by fdncred** [comment]
I'm on Windows today and don't have any issues. Is there a specific dotnet command you run that breaks it? or a specific version?
![image](https://user-images.githubusercontent.com/343840/181773110-3c55428f-4bba-45fa-b98f-18b9189bc1b9.png)

**Comment by SamWindell** [comment]
For me it's with any dotnet command at all. I'll do some more testing for you with this this weekend. See if I can isolate anything on win/mac/linux.

**Comment by fdncred** [comment]
If you're going to do more testing you should probably clone the main branch and build from that. We've had to fix some problems recently.

**Comment by SamWindell** [comment]
It's working! Thanks very much for this.

I was not running the version that I compiled. I was using nushell to replace the nushell binary, which was asking for trouble I admit. 

But nu didn't show an error with the `cp` command when the action was not permitted. I assumed no output meant success. Here's what I mean, `^cp` produces the correct error, where nu's `cp` is silent even though it failed:

![Screenshot from 2022-07-30 09-33-04](https://user-images.githubusercontent.com/15104010/181900551-0212307a-80a8-4f93-9dce-5f6b97cce14b.png)

**Comment by galuszkak** [comment]
Shouldn't this be closed? Looks like it's resolved.

**Comment by fdncred** [comment]
closing. lemme know if it's not resolved and i'll reopen.
